src/rinterface_extra.c

```diff
@@ -8331,6 +8331,59 @@ SEXP R_igraph_incident_edges(SEXP pgraph, SEXP pe, SEXP pmode) {
   return result;
 }
 
+SEXP R_igraph_power_law_fit_new(SEXP data, SEXP xmin, SEXP force_continuous, SEXP pvalue)
+{
+  igraph_vector_t c_data;
+  igraph_plfit_result_t c_res;
+  igraph_real_t c_xmin;
+  igraph_bool_t c_force_continuous, c_compute_pvalue;
+  SEXP result, names;
+
+  SEXP r_result;
+
+  R_SEXP_to_vector(data, &c_data);
+  IGRAPH_R_CHECK_REAL(xmin);
+  c_xmin = REAL(xmin)[0];
+  IGRAPH_R_CHECK_BOOL(force_continuous);
+  c_force_continuous = LOGICAL(force_continuous)[0];
+  IGRAPH_R_CHECK_BOOL(pvalue);
+  c_compute_pvalue = LOGICAL(pvalue)[0];
+
+  IGRAPH_R_CHECK(igraph_power_law_fit(&c_data, &c_res, c_xmin, c_force_continuous));
```

Comment:
We could also call the generated function and use `lengthgets()` to add an element to the returned list. This is better because we then would still rely on autogeneration for the better part of the code here.

Reply:
we can't use generated function. To compure `p-value` we need return value from `igraph_power_law_fit` C function, but we lose this data if use generated function. I think using generated function give to much overhead.

Reply:
Thanks @Antonov548!

@krlmlr do you have further thoughts on this?

Reply:
Thanks, makes sense to me.